conda/utils.py

```diff
@@ -286,3 +288,15 @@ def wrap_subprocess_call(on_win, root_prefix, prefix, command):
         command_args = [shell_path, "-x", script_caller]
 
     return script_caller, command_args
+
+
+def ensure_comspec_set():
+    if 'COMSPEC' not in environ:
+        cmd_exe = join(environ['SystemRoot'], 'System32', 'cmd.exe')
```

Comment:
Is a guard/default needed if 'SystemRoot' or 'windir' are not in environ?

Reply:
Not sure what other guards should be put in place here other than a hard code to cmd.exe.... I found this article with default windows environment variables https://ss64.com/nt/syntax-variables.html. 
Maybe there should also be a check for `join(environ['SystemRoot'], 'SysWOW64', 'cmd.exe')`

Reply:
I think Jonathan is more referring to using something like .get() on environ, so that we don't crash out with a keyerror.  I think that's a good idea here.

Reply:
ahh, gotcha